irohad/ordering/impl/ordering_service_impl.cpp

```diff
@@ -33,55 +33,99 @@ namespace iroha {
         size_t delay_milliseconds,
         std::shared_ptr<network::OrderingServiceTransport> transport,
         std::shared_ptr<ametsuchi::OrderingServicePersistentState>
-            persistent_state)
+            persistent_state,
+        bool is_async)
         : wsv_(wsv),
           max_size_(max_size),
           delay_milliseconds_(delay_milliseconds),
           transport_(transport),
-          persistent_state_(persistent_state),
-          is_finished(false) {
-      updateTimer();
+          persistent_state_(persistent_state) {
       log_ = logger::log("OrderingServiceImpl");
 
       // restore state of ordering service from persistent storage
-      proposal_height = persistent_state_->loadProposalHeight().value();
+      proposal_height_ = persistent_state_->loadProposalHeight().value();
+
+      rxcpp::observable<ProposalEvent> timer =
+          rxcpp::observable<>::interval(
+              std::chrono::milliseconds(delay_milliseconds_),
+              rxcpp::observe_on_new_thread())
+              .map([](auto) { return ProposalEvent::kTimerEvent; });
+
+      if (is_async) {
+        handle_ =
+            rxcpp::observable<>::from(timer, transactions_.get_observable())
```

Comment:
I guess it can be moved to function that accepts callback, so might look like:
```cpp
if (is_async)
  subs([](auto s) {
    return s.merge(rxcpp::synchronize_new_thread())
  })
else
  subs([](auto s) {
    return s.merge()
  };
```